### PR TITLE
0.4 release blog post

### PR DESCRIPTION
**Pull request by byorgey**
Draft blog post highlighting cool features in the newest release.  Comments & suggestions welcome!

Also fix the version number in the `.cabal` file from `0.4` to `0.4.0.0`.  It's too late for the release (it always reports there is a new version available since `0.4.0.0` does not match `0.4`) but oh well.

### CODE REVIEW

File: docs/blog/2023-08-swarm-0.4-release.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    profile = wp
+    tags = Swarm, game, robot, programming, resource
+    categories = Haskell, projects
+
+The [Swarm](https://github.com/swarm-game/swarm/) development team is
+very proud to announce the latest release of the game.  This should
+still be considered a development/preview release---you still can't
+[save your games](https://github.com/swarm-game/swarm/issues/50),
+though we're making progress towards that---but it's made some
+remarkable progress and there are lots of fun things to try.  We
```
Comment (kostmo) [inline docs/blog/2023-08-swarm-0.4-release.md:15]: The word "progress" is repeated.  Perhaps replace one with "headway"?

File: docs/blog/2023-08-swarm-0.4-release.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+always [welcome any new contributors][contrib]---from playtesting, to
+designing new scenarios, to hacking on the Haskell codebase, all
```
Comment (kostmo) [inline docs/blog/2023-08-swarm-0.4-release.md:17]: Use a "closing" long dash
```suggestion
designing new scenarios, to hacking on the Haskell codebase---all
```

File: docs/blog/2023-08-swarm-0.4-release.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+![](../../images/crafting.png)
+
+But it also means that we can now validate all code examples and
+entity names, and even test that the tutorial is pedagogically sound:
+any command used in a tutorial solution must be mentioned in a previous
+tutorial, or else our CI fails!
+
+There are also a number of other small UI improvements, such as
+improved type error messages, inventory search, and a collapsible REPL
```
Comment (kostmo) [inline docs/blog/2023-08-swarm-0.4-release.md:105]: "improve-" is duplicated.  How about "enhanced" for one instance?

**Review by kostmo — APPROVED** [review]

File: docs/blog/2023-08-swarm-0.4-release.md (hunk 2 above)
Comment (byorgey) [reply docs/blog/2023-08-swarm-0.4-release.md:17]: I just removed the entire last sentence.  Editing via deleting fluff is always so satisfying.

**Comment by byorgey** [comment]
Nice to see our mergify rules are working properly and it didn't bother running the test suite.

**Comment by kostmo** [comment]
Should be fun to watch this after posting to Reddit again: https://star-history.com/#swarm-game/swarm

**Comment by byorgey** [comment]
@kostmo Did you post it to Reddit?  I don't see it there.  Or were you just saying we should?

**Comment by kostmo** [comment]
> @kostmo Did you post it to Reddit? I don't see it there. Or were you just saying we should?

Saying we should :)

**Comment by xsebek** [comment]
> Also fix the version number in the .cabal file from 0.4 to 0.4.0.0. It's too late for the release (it always reports there is a new version available since 0.4.0.0 does not match 0.4) but oh well.

Ugh. I should have programmed it more defensively.

**Pull request merged**